src/pprl/matching/local.py

```diff
@@ -0,0 +1,46 @@
+"""Functions for performing matching locally."""
```

Comment:
To my taste, the functions in 'local.py' are too short and not used enough times to be worth wrapping up into a separate module. There's a readability trade-off: on the one hand it's tidy, but on the other hand it becomes much less linear; it requires the reader to search around to follow the underlying logic. I'd negotiate this trade-off differently, but you have the final say.

Reply:
I can see that for sure.

My view is that because the cloud functions get taken out (and they should) these should as well. Creating a module for each behaviour creates consistency, which is no bad thing.

Also, if we don't take them out, then the server script has stuff in it that isn't relevant for most users should they go delving into it.